torchci/clickhouse_queries/pr_commits/query.sql

```diff
@@ -8,37 +7,42 @@ WITH
 -- Get all PRs that were merged into master, and get all the SHAs for commits from that PR which CI jobs ran against
 -- We need the shas because some jobs (like trunk) don't have a PR they explicitly ran against, but they _were_ run against
 -- a commit from a PR
-pr_shas AS (
+shas as (
   SELECT DISTINCT
-    FORMAT_ISO8601(
-        PARSE_TIMESTAMP_ISO8601(p.head_commit.timestamp),
-        'America/Los_Angeles'
-    ) as timestamp,
-    r.pull_requests[1].number AS pr_number,
-    p.head_commit.id AS sha,
-    p.head_commit.message,
+    r.pull_requests[1].'number' AS pr_number,
+    r.head_sha AS sha,
     CONCAT(
       'https://github.com/',
-      :owner,
+      {owner: String},
       '/',
-      :repo,
+      {repo: String},
       '/pull/',
-      r.pull_requests[1].number
-    ) AS pr_url,
-    p.head_commit.url AS commit_url,
+      r.pull_requests[1].'number'
+    ) AS pr_url
   FROM
-    commons.push p
-    JOIN commons.workflow_run r ON p.head_commit.id = r.head_sha HINT(join_strategy=lookup)
+    default.workflow_run r final
   WHERE
-    1 = 1
-    AND LENGTH(r.pull_requests) = 1
-    AND r.repository.owner.login = :owner
-    AND r.pull_requests[1].head.repo.name = :repo
-    AND r.pull_requests[1].number = :pr_num
-
+    LENGTH(r.pull_requests) = 1
+    AND r.repository.'owner'.'login' = {owner: String}
+    AND r.pull_requests[1].'head'.'repo'.'name' = {repo: String}
+    AND r.pull_requests[1].'number' = {pr_num: Int64}
+    and r.id in (select id from materialized_views.workflow_run_by_pr_num where pr_number = {pr_num: Int64})
+),
+shas_with_info AS (
+  SELECT DISTINCT
+    p.head_commit.'timestamp' as timestamp,
+    s.pr_number AS pr_number,
+    s.pr_url as pr_url,
+    p.head_commit.'id' AS sha,
+    p.head_commit.'message' as message,
+    p.head_commit.url AS commit_url
+  FROM
+    default.push p final
+    JOIN shas s ON p.head_commit.id = s.sha
+  where p.head_commit.'timestamp' in (select timestamp from materialized_views.push_by_sha where id in (select sha from shas))
```

Comment:
Does this where statement here needed?  It looks unusual because it uses `in` with a list of timestamp (not regular id type)

Reply:
the push table is currently ordered by the timestamp and this where clause looks at a different version of the table that is ordered by sha, which makes the query run faster

Reply:
TIL! I guess this worth adding a comment note